### ISSUE TITLE
Bug: [naming-convention] `typeMethod` selector doesn't support function expressions 

### ISSUE THREAD

**Issue by kozlovvski** [typescript-eslint/typescript-eslint#6617]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=4.9.3&sourceType=module&code=C4TwDgpgBACgTgezAZygXigbwFBSgenygGME44JjgoA7AQwFsJlcoEaAxBBACgEoAXFABuCAJYATANyt2AITpwh-dAD4R46dlaEoYmqXKVq9JizwALOjQkAbCF16CNkmZet2ICpVBVp1oq7YAL5SQA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6JgQwFtmBzWsgHsmAN0QtKw9GADa4bDkTRog6JAA08hVgXZIyREjL5V0yEVIBZRPgAWggCYatuyMWiIAZpQAe0ubquUi4KALqagTieqlwc+P6QAMr4VMYAChzIZBzwAMKZiJChIWAAvi7FEOWlQA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA

### Repro Code

```typescript
type Props = {
  // correct names
  onFoo(): void;
  onBar: () => void;

  // incorrect names
  handleFoo(): void;
  handleBar: () => void;
};
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/naming-convention": [
      "error",
      {
        "selector": "typeMethod",
        "prefix": [
          "on"
        ],
        "format": ["StrictPascalCase"]
      }
    ]
  },
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    // ...
  }
}
```

### Expected Result

Both `handleFoo` and `handleBar` should be marked as errors.

### Actual Result

Only `handleFoo` is marked as an error. 

### Additional Info

The `typeMethod` selector should pick up every possible method syntax similarly to the `objectLiteralMethod` selector. Right now only `TSMethodSignature` nodes are picked up by the `typeMethod` selector, whereas it should also pickup `TSPropertySignature` nodes with `typeAnnotation` of `TSFunctionType`. 

If you confirm this is a bug, I would be more than happy to create a PR with the fix 😄 

References in code:
- place where every non-computed `TSPropertySignature` is mapped as `typeProperty` selector: https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/src/rules/naming-convention.ts#L397
- place where every non-computed `typeMethod` selector is mapped: https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/src/rules/naming-convention.ts#L463

**Comment by bradzacher** [comment]
Yup this should be mirroring [the same thing done for class methods](https://typescript-eslint.io/play/#ts=4.9.3&sourceType=module&code=MYGwhgzhAEAKBOB7ADjA3gKGtA9D6wi88ApsAC7QB2YAtiRFtIlQGKKIAUAlAFzQA3RAEsAJtDTQAvkxYAhMPGgBeaDxUA+QSPEBGANwYmeaMKqFiZSjXqNsACzBVRIEuy59tYidKaPnrgpKqurKWkLeBhhS+kA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6JgQwFtmBzWsgHsmAN0QtKw9GADa4bDkTRog6JAA08hVgXZIyREjL5V0yGXgdkyALKJ8AC0EATDVt2Ri0RADNKAD2k5XQ8pdwUAXU0QnB9VLg58IMgAZXwqYwAFKzIOeABhK0RICPCwAF93UohK8qA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA).

**Comment by kozlovvski** [comment]
Okay, I can prepare the PR with fixes then
